components/script/dom/workerglobalscope.rs

```diff
@@ -113,17 +113,11 @@ pub struct WorkerGlobalScope {
     location: MutNullableDom<WorkerLocation>,
     navigator: MutNullableDom<WorkerNavigator>,
 
-    #[ignore_malloc_size_of = "Defined in ipc-channel"]
+    #[ignore_malloc_size_of = "Defined in crossbeam"]
     #[no_trace]
-    /// Optional `IpcSender` for sending the `DevtoolScriptControlMsg`
-    /// to the server from within the worker
-    from_devtools_sender: Option<IpcSender<DevtoolScriptControlMsg>>,
```

Comment:
I think removing this field broke a ton of tests.

Reply:
Looks like this `IpcSender` must be stored to keep the channel alive. This was really unclear from the code, so I've left a comment. This should probably be clarified a bit with another data structure that stores all of the different channels necessary for devtools communication.